lib/systems/click_system.cpp

```diff
@@ -32,6 +32,9 @@ void ClickSystem::update(double) {
             y < bottom && y > top) {
             // The button was clicked
             click->fn();
+            // After the fn has been executed we must break the glass.
+            // Because if the fn destroys the current scene with its click components, and completely destory this loop
```

Comment:
I understand what you mean, but the sentence doesn't sound too great. Would go for something along these lines;
"In case the FN destroys the current scene, including the click component the method is stored in, it completely destroys this loop". Would like to see whether you agree and what you come up with.